mla/injectors.py

```diff
@@ -168,86 +167,87 @@ def __init__(self,
             background_grl = event_model.grl[background_runs]
         else:
-            fully_contained = (event_model.grl['start'] >= start_time-background_window) &\
-                                (event_model.grl['stop'] < start_time)
-            start_contained = (event_model.grl['start'] < start_time-background_window) &\
-                                (event_model.grl['stop'] > start_time-background_window)
+            fully_contained = (
+                event_model.grl['start'] >= start_time - background_window
+            ) & (event_model.grl['stop'] < start_time)
+            start_contained = (
+                event_model.grl['start'] < start_time - background_window
+            ) & (event_model.grl['stop'] > start_time - background_window)
             background_runs = (fully_contained | start_contained)
             if not np.any(background_runs):
-                print("ERROR: No runs found in GRL for calculation of "
-                      "background rates!")
+                print('ERROR: No runs found in GRL for calculation of '
+                      'background rates!')
                 raise RuntimeError
             background_grl = event_model.grl[background_runs]
-        
-        #Find the background rate
+
+        # Find the background rate
         n_background = background_grl['events'].sum()
         n_background /= background_grl['livetime'].sum()
         n_background *= background_time_profile.exposure
         self.n_background = n_background
         self.background_time_profile = background_time_profile
         self.signal_time_profile = signal_time_profile
-        
 
     def inject_background_events(self,
                                  event_model: models.EventModel) -> np.ndarray:
         """Injects background events with specific time profile for a trial.
 
         Args:
-            
+
         Returns:
             An array of injected background events.
         """
         n_background_observed = np.random.poisson(self.n_background)
-        background = np.random.choice(event_model.data, n_background_observed).copy()
-        background['ra'] = np.random.uniform(0, 2*np.pi, len(background))
-        background['time'] = self.background_time_profile.random(size = len(background))
+        background = np.random.choice(event_model.data,
+                                      n_background_observed).copy()
+        background['ra'] = np.random.uniform(0, 2 * np.pi, len(background))
+        background['time'] = self.background_time_profile.random(
+            size=len(background))
 
         return background
-    
-    
-    def inject_signal_events(self, 
-                             source: core.Source,
+
+    def inject_signal_events(self, source: core.Source,
                              trial_preprocessing: np.ndarray,
-                             n_signal_observed:Optional[int] = None,
-                             **kwargs) -> np.ndarray:
```

Comment:
we need to keep the **kwargs here, otherwise **kwargs from trial_generators.generate and produce_trial from core will pass into here and throw unexpected KeyError